src/app/tools/ink_processing.h

```diff
@@ -143,6 +143,18 @@ class CopyInkProcessing : public SimpleInkProcessing<CopyInkProcessing<ImageTrai
       switch (loop->sprite()->pixelFormat()) {
         case IMAGE_RGB: m_color |= rgba_a_mask; break;
         case IMAGE_GRAYSCALE: m_color |= graya_a_mask; break;
+        case IMAGE_INDEXED: {
+          Palette* pal = loop->sprite()->palette(loop->getFrame());
+          color_t c = pal->getEntry(m_color);
+          m_color = (rgba_geta(c) < 255 ?
+                      pal->findBestfit(rgba_getr(c),
+                                       rgba_getg(c),
+                                       rgba_getb(c),
+                                       255,
+                                       -1) :
+                      m_color);
```

Comment:
The issue here (super improbable case but possible) is that we lost the original index of the secondary color, e.g. try to load this file in this PR:
 
[try-to-paint-with-index-2.zip](https://github.com/user-attachments/files/16640741/try-to-paint-with-index-2.zip)

and try to paint with index 2, you will get index 1, and this is the most simple case in indexed color mode (as the "copy" in CopyInkProcessing says): we have to just copy the same index color.

I think indexed mode with transparent colors is a special case were we shouldn't take care in these cases, i.e. the background layer can contain specific indexes that might be transparent/semi-transparent (probably something to fix in the rendering side in a future, or just leave it as it is).